### DISCUSSION TITLE
SAM - Smart Agent Manager

### DISCUSSION THREAD

**Discussion by Mallock** [betalgo/openai#157] [Show and tell]
Hi,
Thank you for providing such a user-friendly library. I greatly appreciate it. I'm happy to report that I've successfully created a simple desktop application using your library. 

When I connected my app to Azure Cognitive Services, the experience was absolutely incredible. Not only was I able to get my app to perform tasks using your library, but it was like having a real conversation! The Text-to-Speech and Speech-to-Text capabilities were amazing. I would talk to my microphone, and the app would respond back to me in a human-like voice. It was like Iron Man chatting with Jarvis! 

The smoothness and fluency of the conversation were so cool. Your library and Azure Cognitive Services really brought my app to life, and made the user experience so much more fun, interactive, and engaging.  Thank you for creating such a fantastic library!

Below text is generated using SAM and your ChatGPT:

SAM (Smart Agent Manager) is an innovative AI agent system that enables seamless communication between multiple agents, each with unique roles. The system is optimized to automate routine tasks, streamline workflow, and facilitate collaboration between agents. SAM is licensed under the permissive MIT License.

Features:

- Create multiple agents with different roles
- Enable communication between agents
- Automate routine tasks and optimize workflow
- Ensure smooth collaboration between agents

Installation:

To install SAM, you can either clone the repository and open the solution in Visual Studio 2022, or use the standalone samsetup installer available on the SAM release page on GitHub.

Usage:

Once the installation is complete, you can start using SAM by running the executable file to start the SAM server. With SAM, you can easily create agents, manage workflows, and optimize routines.

Agent Interface:

- Agent Name: the name of the GPT agent currently in use
- Agent ID: the unique identification number or code for the GPT agent.
- Slave Agent Settings: allow the user to forward responses to another GPT agent based on specific instructions. (NOTE: this is only possible if agents have "Smart agent enabled" checked )
- Message to Slave Agent: a text box for inputting a message to be forwarded to a slave agent before the response.
- Agent Control: provides the user with some controls over the GPT agent
- Smart Agent Enabled: enables message forwarding to slave agents based on defined rules
- Conversation Panel: displays the entire chat conversation between the user and the GPT agent
- Code Panel: displays the code provided by the GPT Agent.

Azure Integration:

SAM integrates with Azure to provide users with voice chat capabilities with their selected SAM agent. By integrating with Azure's Text to Speech and Speech to Text APIs, the user's spoken words are converted into text, and the SAM agent's responses are converted back into spoken words. 

Translator Example:

The Teller agent can be configured to send the story to the Translator agents for translation into multiple languages.

Installation Guide:

Follow the steps outlined by the SAM installation guide to get started with SAM.

Contributions:

We welcome contributions from the community to improve SAM. Please fork the repository and submit a pull request.

SAM is an excellent tool for managing multiple agents with unique roles and streamlining workflow. With voice chat capabilities and integration with Azure, SAM is a versatile and valuable tool for developers.

Almost forgot the link :)
https://github.com/Mallock/sam-smart-agent-manager

**Comment by tonym1995** [comment]
Cloned the repository, built and ran first time. Works well. Only done text chat so far and that worked very well.
Couldn't figure out how to do the Slave Agent though. The slave agent names does not allow you to enter a name and none appear even after creating multiple agents.
Any chance you could do a quick tutorial and show how to get the best out of SAM?
Looks promising. Good Work :-)

**Comment by sammadox** [comment]
Hello I hope that everything is doing well,

Firstly I want to say thank you for all your efforts on maintaining the Betalgo repo.

It's a great work, I always read carefully all notifications I got on my Mailbox.

I also really likes, the SAM library shared with us not so long, it's very useful.

I want to contribute in SAM, I have forked the code using Github Desktop.

Read the "Code of Conduct File", made some few changes to see if I were able to contribute in the project.

Opened it in Visual Studio, seen a lot of things that I want to fix but while trying to pull, I have noticed, that my contributions, were not taken into account, with my personnal account but rather with my business account, eventhought I've forked the repo with my personnal account of a company I was fired from it lately, which makes me demotivated.

Also How can we build the .exe file from the source code ?

Note that is the first time I contribute to a project on Github.

https://github.com/sammadox


Thanks.
________________________________
De : tonym1995 ***@***.***>
Envoyé : mardi 14 mars 2023 00:51
À : betalgo/openai ***@***.***>
Cc : Subscribed ***@***.***>
Objet : Re: [betalgo/openai] SAM - Smart Agent Manager (Discussion #157)


Cloned the repository, built and ran first time. Works well. Only done text chat so far and that worked very well.
Couldn't figure out how to do the Slave Agent though. The slave agent names does not allow you to enter a name and none appear even after creating multiple agents.
Any chance you could do a quick tutorial and show how to get the best out of SAM?
Looks promising. Good Work :-)

**Comment by kayhantolga** [comment]
Thank you for your kind words, @Mallock. It's very motivating to see products that use the library!
I have added your project to the samples section of the Wiki page.

**Comment by sammadox** [comment]
That's a pleasure.

Can you add this project to the sample section of the wiki page:

https://github.com/sammadox/CHAT-GPT-Compiler

CHAT GPT Compiler is a compiler for CHAT GPT, a natural language programming language designed to be more accessible to non-programmers than traditional programming languages. The compiler converts CHAT GPT code into executable machine code that can be run on a computer.

Thanks.
________________________________
De : Tolga Kayhan ***@***.***>
Envoyé : mardi 14 mars 2023 23:37
À : betalgo/openai ***@***.***>
Cc : Sammadox ***@***.***>; Comment ***@***.***>
Objet : Re: [betalgo/openai] SAM - Smart Agent Manager (Discussion #157)


Thank you for your kind words, @Mallock<https://github.com/Mallock>. It's very motivating to see products that use the library!
I have added your project to the samples section of the Wiki page.

**Reply by kayhantolga** [reply]
@sammadox hi, I just saw your message. Does your library use Betalgo.OpenAI ?

**Comment by kayhantolga** [comment]
@Mallock is your library still available? The link is not working anymore.
